### ISSUE TITLE
Spurious floattidf  failures on mips

### ISSUE THREAD

**Issue by alexcrichton** [rust-lang/compiler-builtins#168]
This was coming up every so often on https://github.com/rust-lang-nursery/compiler-builtins/pull/166 so wanted to jot this down before I forgot.

```
---- floattidf stdout ----

	thread 'floattidf' panicked at 'assertion failed: `(left == right)`

  left: `((-56294995342131100,), 14335762557005848176, 14335762557005848160, true)`,

 right: `((-56294995342131100,), 14335762557005848176, 14335762557005848160, false)`', /target/mips-unknown-linux-gnu/debug/build/compiler_builtins-7252d96c089bca74/out/floattidf.rs:10055
```

**Comment by est31** [comment]
Yup, I've got this in #161 as well, but lacked the discipline to open an issue :)

**Comment by mattico** [comment]
Here's the test:

```Rust
static TEST_CASES: &[((i128,), u64)] = &[ /*...*/ ];

#[test]
fn floattidf() {
    for &((a,), b) in TEST_CASES {
        let b_ = __floattidf(a);
        let g_b = to_u64(b_);
        let diff = if g_b > b { g_b - b } else { b - g_b };
        assert_eq!(((a,), b, g_b, true), ((a,), b, g_b, diff <= 1));
    }
}
```

**Comment by AaronKutch** [comment]
This has been fixed by #397
